samza-core/src/main/scala/org/apache/samza/system/chooser/DefaultChooser.scala

```diff
@@ -244,7 +245,13 @@ class DefaultChooser(
   /**
    * Metrics registry to be used when wiring up wrapped choosers.
    */
-  registry: MetricsRegistry = new MetricsRegistryMap) extends MessageChooser with Logging {
+  registry: MetricsRegistry = new MetricsRegistryMap,
+
+  /**
+   * Stores the mapping from SystemStreamPartition to number of tasks consuming from it.
+   * This will be used by the MessageChooser to customize the envelope selection.
+   */
+  sspToTasksCount: Map[SystemStreamPartition, Int] = Map()) extends MessageChooser with Logging {
```

Comment:
I think it is not a very good style to use default values in the arugments. If you don't really need them - please remove them.